content/markdown/04-intro-to-stats/04-08-linear-regression.Rmd

```diff
@@ -0,0 +1,468 @@
+
+You might ask, why square it? The short answer to that question is that squaring it makes a large number of mathematical tools available to us. 
+
+It now follows somewhat logically that we want to find a value for the slope of our line, $\beta_1$ that <strong>minimizes</strong> this sum of the squared residuals. What we are doing here is actually a process called <em>estimation</em> wherein we choose some method that allows us to compare candidate values for $\beta_1$ with other values. Ordinary Least Squares is actually just a specific version of Maximum Likelihood Estimation, which we'll discuss below. 
+
+## Linear Model
+
+As in our previous work, we will certainly NOT (!!!!) just jump into writing code, as that's the best way to ensure a poorly done analysis. First, we have our question. Let's frame it as a hypothesis. It seems logical here to state that we think temperature will positively vary with ice cover. So a null hypothesis may be that <em> there is no relationship between air temperature and ice cover </em>. We have decided we want to use linear regression to tackle this challenge, so consider the assumptions of the method: 
+
+1. Linearity - we assume the relationship between our two variables, $X$ and $Y$ is linear
+
+2. Homoskedacity - we assume the variance of the residual is the same for any value of $X$
+
+3. Normality - We assume that for any fixed value of $X$, the errors of $Y$ normally distributed
+
+4. Randomness - We also assume that samples were collected at random
+
+5. Independence - All errors are independent
+
+If assumptions 1-3 are violated, then we can may be able to use some sort of transformation on our response variable to deal with the problem. 
+
+### Small example 
+
+Let's now get our hands dirty with a bit of an example, a very small one, that we'll go all the way through and then delve into some details in the next example. Very generally, we can think of the process of doing regression as falling into a few steps: 
+
+1) Inspect our data 
+
+2) Perform the regression
+
+3) Assess our assumptions (post-hoc)
+
+4) Inspect our results
+
+5) Interpret & Visualize
+
+
+Here are the data we will start with: 
+
+```{r}
+# set seed -- see section on this in the probability section
+set.seed(1234)
+
+x = c(1:100)
+y = rep(NA, 100)
+for(i in 1:100) {
+  y[i] = 1.3 * x[i] + 10 + rnorm(1, mean = 0, sd = 23)
+}
+
+df = data.frame(
+  x = x, y = y
+)
+
+head(df)
+```
+
+Often it's true when doing regression modelling that it's easier to do our diagnostics of assumptions <em>after</em> we actually fit the model itself as some more simple tools will become available to us. So, let's start with fitting the model. As we mentioned before, R makes it dead simple to do a linear regression in R with the <code>stats::lm()</code> function. 
```

Comment:
Agree, but you can also get an idea of whether the data violates the assumptions by plotting x vs. y to start